Sources/NIO/ContiguousCollection.swift

```diff
@@ -0,0 +1,82 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2017-2018 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+/// A `Collection` that is contiguously layed out in memory and can therefore be duplicated using `memcpy`.
```

Comment:
Nit: s/layed/laid/

Reply:
just moved that line over but obvs fixed. @Lukasa please also re-review the thing as @lorentey had a great idea for writing the conformance for `Slice`